### PR TITLE
Remove deprecated functionality in JAG models

### PR DESCRIPTION
**Pull request by timmoon10**
This is a followup to #667. It primarily removes target layers and reconstruction layers from the JAG models.

### CODE REVIEW

File: model_zoo/models/alexnet/model_alexnet.prototext [hunk 1]
```diff
@@ -1,5 +1,4 @@
 model {
-  name: "directed_acyclic_graph_model"
```
Comment (samadejacobs) [inline model_zoo/models/alexnet/model_alexnet.prototext]: Why deleting this? Is "directed_acyclic_graph_model" default model name in LBANN?

File: model_zoo/models/autoencoder_cifar10/model_autoencoder_cifar10.prototext [hunk 2]
```diff
@@ -47,18 +43,30 @@ model {
   #######
   layer {
     name: "data"
-    children: "encode1 reconstruction"
+    children: "image label"
```
Comment (samadejacobs) [inline model_zoo/models/autoencoder_cifar10/model_autoencoder_cifar10.prototext]: Calling the second child of data layer "label" seems like a misnomer to me. It is not "label", rather a replica of input data, can we find a clearer/meaningful name?

**Comment by samadejacobs** [comment]
Also, the title of this PR is misleading, more files were touched than JAG models. This is a minor thing, so I approve.

File: model_zoo/models/alexnet/model_alexnet.prototext (hunk 1 above)
Comment (timmoon10) [reply model_zoo/models/alexnet/model_alexnet.prototext]: I've changed DAG models to be the default.

**Comment by timmoon10** [comment]
This PR includes PR #667. The diff would be smaller if Github had a nice interface for staged pull requests.

**Pull request merged**